modules/farm/spec/02_messages.md

```diff
@@ -0,0 +1,136 @@
+<!--
+order: 2
+-->
+
+# Messages
+
+## MsgCreatePool
+
+The farm pool can be created by any user via a `MsgCreatePool` message.
+
+```go
+type MsgCreatePool struct {
+    Name           string
+    Description    string
+    LpTokenDenom   string
+    StartHeight    uint64
+    RewardPerBlock sdk.Coins
+    TotalReward    sdk.Coins
+    Destructible   bool
+    Creator        string
+}
+```
+
+This message is expected to fail if:
+
+- `LpTokenDenom` does not comply with the rules specified on the chain.
+- the name of farm pool has exist.
+- `StartHeight` is less than the current block height.
+- `TotalReward` is less than `RewardPerBlock`.
+- the balance of creator is not enough to pay `CreatePoolFee+TotalReward`.
+- The lenght of `TotalReward` is greater than `MaxRewardCategoryN`.
+
+In addition, the `Endheight = TotalReward/RewardPerBlock`.Because there may be multiple tokens for event rewards, the end heights may be inconsistent. In order to reduce the complexity of the system, take the smallest value among all heights as the final end height. After the event ends, the remaining rewards will be refunded To creator's account.
```

Comment:
```suggestion
In addition, the `Endheight = TotalReward/RewardPerBlock`. Because there may be multiple tokens for event rewards, the end heights may be inconsistent. In order to reduce the complexity of the system, take the smallest value among all heights as the final end height. After the event ends, the remaining rewards will be refunded to the creator's account.
```